### PR TITLE
test: Migrate CI to GtiHub [SDK-4450]

### PR DESCRIPTION
**Pull request by evansims**
### 📋 Changes

This pull request replaces our current CircleCI continuous integration suite for auth0-react with a new workflow using GitHub Actions.

### 📎 References

Please see internal ticket SDK-4450.

### 🎯 Testing

This pull request replaces our continuous integration suite with one built in GitHub's native Actions ecosystem. It aligns with other migrations taking place across our repositories.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.inputs.branch }}
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: npm
```
Comment (frederikprijck) [inline .github/workflows/publish.yml:101]: We should set the registry-url to npm here, else it wont use authentication. See:

- https://github.com/actions/setup-node/blob/main/src/main.ts#L59-L61
- https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

We encountered this with lock and fixed it: https://github.com/auth0/lock/pull/2445

**Review by frederikprijck — DISMISSED** [review]

**Comment by evansims** [comment]
@adamjmcgrath I am unable to get `npm test` to work on either the CI migration or my local machine. Could you please take a look?

**Comment by frederikprijck** [comment]
With @adamjmcgrath being off, I'll look into this.

**Comment by evansims** [comment]
@frederikprijck Great, thanks. (FWIW, regarding your last commit, it already builds before running tests; see the `build` job the `unit` job `need`s beforehand.)

**Comment by evansims** [comment]
@adamjmcgrath All migrated tests appear to be passing successfully with the workflows today. Uncertain why they hadn't worked previously, the error messages were rather confusing and vague.

CircleCI test is also randomly busted, it appears, so doesn't look related to anything to do with the migration.

That being the case, PR is ready for your review, thanks!

**Review by adamjmcgrath — CHANGES_REQUESTED** [review]
@evansims - I'm still seeing failed tests even though the "Run Unit Tests" job is reporting as successful 

image

Will take a look at the tests when I have some time

**Comment by adamjmcgrath** [comment]
Unit tests are passing on GH Actions, just need to investigate why Browserstack tests aren't running

**Pull request merged**